openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -1659,6 +1540,155 @@ TCN_IMPLEMENT_CALL(void, SSLContext, setCertVerifyCallback)(TCN_STDARGS, jlong c
     }
 }
 
+/**
+ * Returns an array containing all the X500 principal's bytes.
+ *
+ * Partly based on code from conscrypt:
+ * https://android.googlesource.com/platform/external/conscrypt/+/master/src/main/native/org_conscrypt_NativeCrypto.cpp
+ */
+static jobjectArray principalBytes(JNIEnv* e, const STACK_OF(X509_NAME)* names) {
+    jobjectArray array;
+    jbyteArray bArray;
+    int i;
+    int count;
+    int length;
+    unsigned char *buf;
+    X509_NAME* principal;
+
+    if (names == NULL) {
+        return NULL;
+    }
+
+    count = sk_X509_NAME_num(names);
+    if (count <= 0) {
+        return NULL;
+    }
+
+    array = (*e)->NewObjectArray(e, count, byteArrayClass, NULL);
+    if (array == NULL) {
+        return NULL;
+    }
+
+    for (i = 0; i < count; i++) {
+        principal = sk_X509_NAME_value(names, i);
+        buf = NULL;
+        length = i2d_X509_NAME(principal, &buf);
+        if (length < 0) {
+            if (buf != NULL) {
+                // We need to delete the local references so we not leak memory as this method is called via callback.
+                OPENSSL_free(buf);
```

Comment:
@nmittler - I did some code inspection (referenced in https://github.com/netty/netty-tcnative/pull/164#discussion_r68122457) and IIUC this can be set to `free`. `free` does not have the same `NULL` protections offered by `delete`.